app/tests/spec/lib/experiments/mailcheck.js

```diff
@@ -1,143 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-define(function (require, exports, module) {
-  'use strict';
-
-  const Able = require('lib/able');
-  const chai = require('chai');
-  const Experiment = require('lib/experiments/mailcheck');
-  const Metrics = require('lib/metrics');
-  const Notifier = require('lib/channels/notifier');
-  const sinon = require('sinon');
-  const Storage = require('lib/storage');
-  const TestHelpers = require('../../../lib/helpers');
-  const User = require('models/user');
-  const WindowMock = require('../../../mocks/window');
-
-  var able;
-  var assert = chai.assert;
-  var experiment;
-  var notifier;
-  var metrics;
-  var storage;
-  var user;
-  var UUID = 'a mock uuid';
-  var windowMock;
-
-  describe('lib/experiments/mailcheck', function () {
-    beforeEach(function () {
-      able = new Able();
-      sinon.stub(able, 'choose', function () {
-        return 'treatment';
-      });
-      notifier = new Notifier();
-      metrics = new Metrics({ notifier });
-      storage = new Storage();
-      user = new User({
-        uniqueUserId: UUID
-      });
-      windowMock = new WindowMock();
-
-      experiment = new Experiment();
-      experiment.initialize('mailcheck', {
-        able: able,
-        metrics: metrics,
-        notifier: notifier,
-        storage: storage,
-        user: user,
-        window: windowMock
-      });
-    });
-
-    describe('initialize', function () {
-      it('initializes', function () {
-        assert.isTrue(experiment._initialized);
-      });
-    });
-
-    describe('_onSignupSubmit', function () {
-      it('logs the corrected event', function () {
-        var mockView = {
-          $el: {
-            find () {
-              return {
-                val () {
-                  return 'a@gmail.com';
-                },
-                data () {
-                  return 'a@gmail.com';
-                }
-              };
-            }
-          }
-        };
-        notifier.trigger('signup.submit', {}, mockView);
-        assert.isTrue(TestHelpers.isEventLogged(metrics, 'experiment.treatment.mailcheck.corrected'));
```

Comment:
@divyabiyani ok perfect, then we can just clean up the code in (so it lives in its own function ) https://github.com/mozilla/fxa-content-server/pull/4703#discussion_r102228528 and it will already be tested 👍